From b1a36ceb3bb0c8de45fc2024e57529e02ee3adef Mon Sep 17 00:00:00 2001
From: Andreas Jaeger <aj@suse.de>
Date: Mon, 29 Apr 2013 21:00:57 +0200
Subject: [PATCH 1/4] BZ#15086: Fix res_debug printing of options

	[BZ# 15086]
	* resolv/res_debug.c (p_option): Handle RES_NOALIASES,
	RES_KEEPTSIG, RES_BLAST, RES_NOIP6DOTINT, RES_SNGLKUP,
	RES_SNGLKUPREOP.
---
 ChangeLog          | 7 +++++++
 NEWS               | 8 ++++----
 resolv/res_debug.c | 6 ++++++
 3 files changed, 17 insertions(+), 4 deletions(-)

diff --git a/resolv/res_debug.c b/resolv/res_debug.c
index 3daa44e..c064d0c 100644
--- a/resolv/res_debug.c
+++ b/resolv/res_debug.c
@@ -582,11 +582,17 @@ p_option(u_long option) {
 	case RES_DNSRCH:	return "dnsrch";
 	case RES_INSECURE1:	return "insecure1";
 	case RES_INSECURE2:	return "insecure2";
+	case RES_NOALIASES:	return "noaliases";
 	case RES_USE_INET6:	return "inet6";
 	case RES_ROTATE:	return "rotate";
 	case RES_NOCHECKNAME:	return "no-check-names";
+	case RES_KEEPTSIG:	return "keeptsig(unimpl)";
+	case RES_BLAST:		return "blast";
 	case RES_USEBSTRING:	return "ip6-bytstring";
+	case RES_NOIP6DOTINT:	return "no-ip6-dotint";
 	case RES_USE_EDNS0:	return "edns0";
+	case RES_SNGLKUP:	return "single-request";
+	case RES_SNGLKUPREOP:	return "single-request-reopen";
 	case RES_USE_DNSSEC:	return "dnssec";
 	case RES_NOTLDQUERY:	return "no-tld-query";
 				/* XXX nonreentrant */
-- 
1.8.2.3

